### DISCUSSION TITLE
Simple way to restart devices if they reset.

### DISCUSSION THREAD

**Discussion by iotinkognito** [make-all/tuya-local#1090] [Q&A]
I have a number of down lights that if they are physically switched off don't reconnect locally unless I restart them individually. I am wondering is there an easy way to restart the entire integration and all entities etc.? 

I am also thinking trying to to template a automation that if any Tuya local device is disconnected and reconnected to my ubiquiti wifi network it will automatically restart that tuyalocal integration.

Thoughts?

**Comment by make-all** [comment]
Home Assistant has the ability to restart an integration.  However, there are issues with hung connections if your devices are not responding.

Best solution is to not turn off your smart devices.

Setting the IP address and protocol version explicitly will also help them recover quicker than if they are set to Auto.

**Comment by iotinkognito** [comment]
Yeah but you seem to have to do that restart on each device.

They have manually set ip addresses. I have manually added them to HA for testing before I switch over my 130 Tuya devices.

They still respond from cloud through Smart life or Tuya Cloud, it's like they also maintain their local connection but can no longer be controlled from HA unless I goto settings -> devices and services -> Tuya Local - > Scroll to devices click the burger menu and press reload. After that it's all good almost instantly.

Not turning them off is an option but not a solution for off-grid the milliwatts add up!


________________________________

**Reply by iotinkognito** [reply]
This error originated from a custom integration.

Logger: custom_components.tuya_local.device
Source: custom_components/tuya_local/device.py:222
Integration: Tuya Local (documentation, issues)
First occurred: 8:53:12 AM (1 occurrences)
Last logged: 8:53:12 AM

DL Pantry LOCAL receive loop has terminated

**Reply by iotinkognito** [reply]
Thats the error I get when it would be good to catch and restart the integration

**Reply by make-all** [reply]
The only reason the receive loop terminates is if the integration has received a shutdown notification from HA.  It would be good to understand why that happens (and it certainly would not be correct to try to restart the integration when HA is trying to shutdown).

**Comment by iotinkognito** [comment]
Perhaps I found the wrong error there. I have two downlights, both with fixed addresses. The downlights are on a physical light switch. If I turn off the switch the lights only sometimes don't require a restart on the individual device to be controlled by Tuya Local.  When not working in Tuya Local they always work via cloud integration or via smart life. If I go to the device itself and click reload it always comes back. I was hoping there is a simple way to just restart the whole integration rather than having to go to each individual device and manually reloading them.

[Image]



Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
